src/WebformCivicrmPostProcess.php

```diff
@@ -2523,7 +2528,11 @@ private function fillDataFromSubmission() {
           }
         }
         elseif ($dataType == 'File') {
-          if (empty($val[0]) || !($val = $this->saveDrupalFileToCivi($val[0]))) {
+          // Replace filename (with tokens) if set.
+          if (isset($component['#file_name']) && $component['#file_name']) {
+            $newFilename = $this->handler->tokenManager->replace($component['#file_name'], $this->submission);
+          }
+          if (empty($val[0]) || !($val = $this->saveDrupalFileToCivi($val[0], $newFilename))) {
```

Comment:
I've just hit an error here with a vanilla install and I think this is the problem. It assumes we set the new Filename above. But I'm getting a warning "Undefined variable $newFilename " which suggests we didn't.